javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -313,6 +363,17 @@ public void shouldSerializeDeserializeNonEmptyMap() {
         assertThat(actual).isEqualTo(expected);
     }
 
+    // -- fill
+
+    @Test
+    public void shouldFillWithComparator() {
+        final LinkedList<Integer> ints = new LinkedList<>(asList(0, 0, 1, 1, 2, 2));
+        final Supplier<Tuple2<Long, Float>> supplier = () -> Tuple.of(ints.remove().longValue(), ints.remove().floatValue());
```

Comment:
@paplorinc in fact, a supplier seems a bit unnecessary for the fill API. Will put it on the list for 3.0.0 to make it an eager evaluated method parameter instead of a supplied value.